### ISSUE TITLE
[ANSIBLE0012] and [ANSIBLE0011]  bug report ansible-lint

### ISSUE THREAD

**Issue by vibhor995** [ansible/ansible-lint#336]
# issue-type
- Bug report

# Ansible and Ansible Lint details

ansible --version 2.5.0

ansible-lint --version 3.4.21
```

- ansible installation method:  pip
- ansible-lint installation method:  pip, 



# Actual Behaviour (Bug report only)

Please give some details of what is actually happening.
Include a [minimum complete verifiable example](http://stackoverflow.com/help/mcve)
with:      

**[ANSIBLE0012] Commands should not change things if nothing needs doing**

ansible-lint is giving error for shell module, even if script is formatted and correct 

**output of ansible-lint**

[ANSIBLE0012] Commands should not change things if nothing needs doing
/home/vibhor/openebs/e2e/ansible/playbooks/feature/ha/ha-cleanup.yml:5
Task/Handler: Delete percona mysql pod

[ANSIBLE0012] Commands should not change things if nothing needs doing
/home/vibhor/openebs/e2e/ansible/playbooks/feature/ha/ha-cleanup.yml:11
Task/Handler: Confirm percona pod has been deleted




and

**[ANSIBLE0011] All tasks should be named**
ansible-lint is giving  this report for set_fact flag=pass which is actually does not require - name module

**output**

[ANSIBLE0011] All tasks should be named
ha.yml:205
Task/Handler: set_fact flag=Pass __file__=ha.yml __line__=206

[ANSIBLE0011] All tasks should be named
ha.yml:209
Task/Handler: set_fact flag=Fail __file__=ha.yml __line__=210

**# expected behaviour** 

clean ansible-lint output , without any ansible-lint rule report like above.



- if you're getting a stack trace, output of
  `ansible-playbook --syntax-check playbook`
 [WARNING]: provided hosts list is empty, only localhost is available. Note that the implicit localhost does not match 'all'


playbook: ha.yml

**Comment by willthames** [comment]
This is by no means a minimal example - please reduce this report to a reasonable sized test case.

I'm not sure what you're expecting to happen vs what is actually happening

**Comment by vibhor995** [comment]
sorry for that long post as i am new git user ,please help me with this issue.

**Comment by willthames** [comment]
Please provide a very simple example of what is happening (one or two tasks, not an entire playbook) and what you expect to happen vs what is actually happening

**Comment by vbotka** [comment]
**[ANSIBLE0012] Commands should not change things if nothing needs doing**
+1

There is no reason for a Warning (severity:High!) . The command is executed only if there are items in the list.

````
> ansible-lint freebsd-postinstall.yml
[ANSIBLE0012] Commands should not change things if nothing needs doing
/home/vlado/.ansible/roles/vbotka.freebsd_postinstall/tasks/mount-iso.yml:78
Task/Handler: mount-iso: Delete nodes
````

````
- name: "mount-iso: Delete nodes"                                                                                                                                                             
  command: "mdconfig -d -u {{ item }}"                                                                                                                                                        
  loop: "{{ local_node_list|default([]) }}"
````

**Comment by awcrosby** [comment]
Regarding the original issue text... for a task that uses a command, even if the script is formatted and correct, it is recommended to use `changed_when`, see this [blog post](https://www.middlewareinventory.com/blog/ansible-changed_when-and-failed_when-examples/) for examples.

If this simply isn’t necessary for your task, you can direct ansible-lint to skip this task, see [docs](https://docs.ansible.com/ansible-lint/rules/rules.html#false-positives-muting-ansible-lint-warnings) (`skip_ansible_lint` will skip to all ansible-lint rules for that task - in the next minor release we plan to allow for more granular skipping - to skip a particular rule for a particular task).

**Comment by vbotka** [comment]
Thank you for the comment. 

For the record. No warning after **changed_when** was added to the task.

    - name: "mount-iso: Delete nodes"
      command: "mdconfig -d -u {{ item }}"
      loop: "{{ local_node_list|default([]) }}"
      changed_when: local_node_list|length > 0
